hkmc2/shared/src/test/mlscript/flows/LeadingDotAccesses.mls

```diff
@@ -0,0 +1,151 @@
+:js
+:flow
+
+class A with
+  fun b = new A
+  fun c = (x) => new A
+module A with
+  fun a = new A
+fun f(x : A) = x
+
+:sf
+f(.a)
+//│ Flowed:
+//│   f⁰(_?_.a{ ~> A⁰.a⁰ })‹app⁰›
+//│ where
+//│     app⁰  <~  type A
+//│ = A
+
+:sf
+f(.a.b)
+//│ Flowed:
+//│   f⁰(A⁰.a⁰.b⁰)‹app¹›
+//│ where
+//│     app¹  <~  type A
+//│ = A
+
+:sf
+let z = .a
+f(z)
+//│ Flowed:
+//│   let z⁰,
+//│   z⁰ = _?_.a{ ~> A⁰.a⁰ },
+//│   f⁰(z⁰)‹app²›
+//│ where
+//│     z⁰  <~  _?_.a
+//│     z⁰  ~>  type A
+//│     app²  <~  type A
+//│ = A
+//│ z = A
+
+:sf
+let z = .a.b.c(3)
+f(z).c(4).b
+//│ Flowed:
+//│   let z¹,
+//│   z¹ = A⁰.a⁰.b⁰.c⁰(3)‹app³›,
+//│   f⁰(z¹)‹app⁴›.c⁰(4)‹app⁵›.b⁰
+//│ where
+//│     z¹  <~  _?_.a
+//│     z¹  ~>  type A
+//│     app³  <~  A
+//│     app³  ->  bind⁰
+//│ = A
+//│ z = A
+
+:sf
+f(.a.c(1).b.c(4))
+//│ Flowed:
+//│   f⁰(A⁰.a⁰.c⁰(1)‹app⁶›.b⁰.c⁰(4)‹app⁷›)‹app⁸›
+//│ where
+//│     app⁸  <~  type A
+//│     app⁷  <~  A
+//│     app⁷  ->  bind¹
+//│ = A
+
+class C
+module C with
+  fun a = new C
+
+:sf
+:ge
+:e
+fun g(x : C) = x
+let d = .a
+f(d)
+g(d)
+//│ ╔══[ERROR] Ambiguous selection with multiple apparent targets:
+//│ ║  l.74: 	let d = .a
+//│ ║        	        ^^
+//│ ╟── companion member a
+//│ ║  l.68: 	  fun a = new C
+//│ ║        	  ^^^^^^^^^^^^^
+//│ ╟── companion member a
+//│ ║  l.8: 	  fun a = new A
+//│ ╙──     	  ^^^^^^^^^^^^^
+//│ Flowed:
+//│   fun g⁰(x⁰: C⁰) ‹flow:g⁰› = x⁰,
+//│   let d⁰,
+//│   d⁰ = _?_.a,
+//│   f⁰(d⁰)‹app⁹›,
+//│   g⁰(d⁰)‹app¹⁰›
+//│ where
+//│     x⁰  <~  type C
+//│     x⁰  ->  app¹⁰
+//│     d⁰  <~  _?_.a
+//│     d⁰  ~>  type A  type C
+//│     app⁹  <~  type A
+//│     app¹⁰  <~  type C
+//│     flow:g⁰  <~  ((type C) -> x⁰)
+//│     flow:g⁰  ~>  ((d⁰) -> app¹⁰)
+//│ ╔══[COMPILATION ERROR] Unexpected term form in expression position (leading dot selection)
+//│ ║  l.74: 	let d = .a
+//│ ╙──      	        ^^
+//│ ╔══[COMPILATION ERROR] No definition found in scope for member 'd'
+//│ ╟── which references the symbol introduced here
+//│ ║  l.74: 	let d = .a
+//│ ╙──      	    ^
+//│ ═══[RUNTIME ERROR] ReferenceError: d is not defined
+
+:sf
+f(.a)
+g(.a)
+//│ Flowed:
+//│   f⁰(_?_.a{ ~> A⁰.a⁰ })‹app¹¹›,
+//│   g⁰(_?_.a{ ~> C⁰.a¹ })‹app¹²›
+//│ where
+//│     app¹¹  <~  type A
+//│     app¹²  <~  type C
+//│ = C
+
+:sf
+:ge
+:re
+:e
+f(.w)
+//│ ╔══[ERROR] Cannot resolve leading dot selection
+//│ ║  l.125: 	f(.w)
+//│ ╙──       	  ^^
+//│ Flowed:
+//│   f⁰(_?_.w)‹app¹³›
+//│ where
+//│     app¹³  <~  type A
+//│ ╔══[COMPILATION ERROR] Unexpected term form in expression position (leading dot selection)
+//│ ║  l.125: 	f(.w)
+//│ ╙──       	  ^^
+//│ ═══[RUNTIME ERROR] Error: Function 'f' expected 1 argument but got 0
```

Comment:
This is similar to the previous comment. I think we won't want to raise an error `Unexpected term form in expression position (leading dot selection)` that is directly caused by another error `Cannot resolve leading dot selection`. This would create duplicate errors and unmeaningful errors to users.

The probably right way is to expand a term to `Term.Error` to silence its lowering error, given that we have already raised an error earlier regarding the same thing.